### ISSUE TITLE
Aern reraise malfunctions

### ISSUE THREAD

**Issue by Brierre** [DerpyProjectGroup/topaz#369]
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] I have searched existing [issues](https://github.com/LandSandBoat/server/issues) to see if the issue has already been opened
- [x] I have checked the commit log to see if the issue has been resolved since my server was last updated
- [x] I have read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)
- [x] I have specified what branch this happens on branch: Base

**_Additional Information_** (Steps to reproduce/Expected behavior) **:** 

There are a couple of different problems here, both seemingly caused by the mixin code not being able to identify a target when it attempts to reraise the mob:

If you kill the aern and it is slated to reraise using the aern family mixin, then
  1. If you run out of range (40 yalms), the reraise "happens" but the mob animation never resets and it is untargetable/useless from that point forward. They still go through the motions of being alive, but you can't see/target them.

  2. If you are killing with a pet, and the pet dies or is released in that time frame when it is applying the reraise code, the same thing happens. On our server when we were still based on Topaz, it was crashing the whole server when it couldn't identify a target in this instance, so we fixed this one by adding if target:isPet() then target = target:getMaster() end to the mixin. It would then attack the player upon reraising. With the new way things work, I'm not sure if that fix would be appropriate.

**Comment by Era-Lusiphur** [comment]
So I wonder about the Aerns not being visible. I think another similar/related possible bug is "mount sickness" as mount users sometimes call it. If you're in a party with other mount users and they dismount before you arrive at a rendezvous point, quite often they will not be visible unless they re-mount and dismount.

**Comment by Brierre** [comment]
You can see their dead name moving around, as if they're roaming normally, but it's like trying to target a dead body.

**Comment by Era-Lusiphur** [comment]
Interesting. In the case of my players and mounts issue, there's no names or anything.

**Comment by Snugglesffxi** [comment]
fwiw if u cast a ga spell next to them it'll hit them and their "name" will become targetable/killable, at least something to tell anyones playerbase incase they have a pile of dead aerns they need to kill 🤣

**Comment by TeoTwawki** [comment]
> 
> 
> fwiw if u cast a ga spell next to them it'll hit them and their "name" will become targetable/killable, at least something to tell anyones playerbase incase they have a pile of dead aerns they need to kill rofl

yep this works because when they get hit the server send the client an update packet again, renewing their status in the client.
